sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCWriteSuite.scala

```diff
@@ -44,9 +44,6 @@ class JDBCWriteSuite extends SharedSQLContext with BeforeAndAfter {
 
   val testH2Dialect = new JdbcDialect {
     override def canHandle(url: String) : Boolean = url.startsWith("jdbc:h2")
-    override def getCatalystType(
```

Comment:
First of all, it's my bad that I didn't notice that there were two `testH2Dialect`.

Now, I understand that this is intentional for the specific test case `Remap types via JdbcDialects`, but it seems to be only valid for `JDBCSuite.scala`.

I mean we can use a normal JdbcDialect in `JDBCWriteSuite` because it's a different suite. Actually, while making #15664, I got stuck by this bug.
Is it possible to fix this correctly in `JDBCWriterSuite`?